### PR TITLE
Transaction history

### PR DESCRIPTION
**Pull request by Nosherwan-Akram**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -8,6 +8,8 @@ import {
 
 import { CssBaseline } from "@mui/material";
 import Sidebar from "components/sidebar/sidebar.component";
+import { TransactionHistory } from "pages/transaction-history/transaction-history.page";
+import { Bills } from "pages/bills/bills.page";
 import Dashboard from "pages/dashboard/dashboard.page";
 import { Transfer } from "transfer/transfer.page";
 // import './App.css';
```
Comment (omerzn) [inline src/App.tsx]: remove this comment

File: src/pages/bills/bills.page.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  };
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const [bills, setBills] = useState([
+    {} as {
```
Comment (omerzn) [inline src/pages/bills/bills.page.tsx]: Please use an interface here as well, there should be uniformity in the project.

File: src/pages/bills/bills.page.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                Add Bill
+              </Button>
+            </Grid>
+            <Divider />
+            <br />
+            <TableContainer component={Paper}>
+              <Table>
+                <TableHead>
+                  <TableRow>
+                    <StyledTableCell align="center">Bill Type</StyledTableCell>
```
Comment (omerzn) [inline src/pages/bills/bills.page.tsx:102]:
```suggestion
                    <StyledTableCell align="center">
                         Bill Type
                     </StyledTableCell>
```

File: src/pages/bills/bills.page.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                    <StyledTableCell align="center">
+                      Reference Number
+                    </StyledTableCell>
+                    <StyledTableCell align="center">Amount</StyledTableCell>
```
Comment (omerzn) [inline src/pages/bills/bills.page.tsx:106]: Format all of the table cells like the Reference Number one.

File: src/pages/bills/bills.page.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                      Late Submission Amount
+                    </StyledTableCell>
+                    <StyledTableCell align="center">Status</StyledTableCell>
+                    <StyledTableCell align="center">Due Date</StyledTableCell>
+                    <StyledTableCell align="center">Actions</StyledTableCell>
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {bills.map((bill) => (
+                    <StyledTableRow key={bill.billType}>
```
Comment (omerzn) [inline src/pages/bills/bills.page.tsx]: is `bill.billType` a unique identifier?

File: src/pages/bills/components/popup.component.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { Divider } from "@mui/material";
+
+interface popupProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+export const Popup = ({ open, onClose }: popupProps) => {
+  return (
+    <div>
```
Comment (omerzn) [inline src/pages/bills/components/popup.component.tsx:41]: is this div necessary?

File: src/pages/transaction-history/transaction-history.page.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  };
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const [transactions, setTransactions] = useState([
+    {} as {
```
Comment (omerzn) [inline src/pages/transaction-history/transaction-history.page.tsx]: interface?

File: src/pages/transaction-history/transaction-history.page.tsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+      transactionAccountTitle: string;
+      balanceAfterTransaction: number;
+    },
+  ]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      setLoading(false);
+      setTransactions(users[0].transactions.reverse());
```
Comment (omerzn) [inline src/pages/transaction-history/transaction-history.page.tsx:48]: there is library that Abdul Basit is using to fetch data from constants file, it mocks api logic and flow. I think if you can't implement the backend, maybe we can use that instead?

File: src/pages/transaction-history/transaction-history.page.tsx [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+              <Table>
+                <TableHead>
+                  <TableRow>
+                    <StyledTableCell align="center">
+                      Account Title
+                    </StyledTableCell>
+                    <StyledTableCell align="center">
+                      Account Number
+                    </StyledTableCell>
+                    <StyledTableCell align="center">Amount</StyledTableCell>
```
Comment (omerzn) [inline src/pages/transaction-history/transaction-history.page.tsx:81]: proper formatting

File: src/pages/transaction-history/transaction-history.page.tsx [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+                    <StyledTableCell align="center">
+                      Transaction Type
+                    </StyledTableCell>
+                    <StyledTableCell align="center">
+                      New Balance
+                    </StyledTableCell>
+                    <StyledTableCell align="center">Date</StyledTableCell>
```
Comment (omerzn) [inline src/pages/transaction-history/transaction-history.page.tsx:88]: here as well

**Review by omerzn — CHANGES_REQUESTED** [review]
Please take a look at the requested changes

File: src/pages/transaction-history/transaction-history.page.tsx (hunk 8 above)
Comment (Nosherwan-Akram) [reply src/pages/transaction-history/transaction-history.page.tsx:48]: I have replaced all this with context and useReducer in the next PR and that acts like a global state. Won't need to use a library.

File: src/pages/bills/bills.page.tsx (hunk 3 above)
Comment (Nosherwan-Akram) [reply src/pages/bills/bills.page.tsx:102]: ESlint and prettier autosaves this Omer bhai

File: src/pages/bills/bills.page.tsx (hunk 4 above)
Comment (Nosherwan-Akram) [reply src/pages/bills/bills.page.tsx:106]: explained above

File: src/pages/bills/bills.page.tsx (hunk 5 above)
Comment (Nosherwan-Akram) [reply src/pages/bills/bills.page.tsx]: I noticed this problem and resolved it on the next PR should I do it here too?

File: src/pages/transaction-history/transaction-history.page.tsx (hunk 7 above)
Comment (omerzn) [reply src/pages/transaction-history/transaction-history.page.tsx]: @Nosherwan-Akram is this done?

File: src/pages/transaction-history/transaction-history.page.tsx (hunk 8 above)
Comment (omerzn) [reply src/pages/transaction-history/transaction-history.page.tsx:48]: That's for backend logic

File: src/pages/transaction-history/transaction-history.page.tsx (hunk 7 above)
Comment (Nosherwan-Akram) [reply src/pages/transaction-history/transaction-history.page.tsx]: Yes

**Pull request merged**